packages/build/src/error/types.ts

```diff
@@ -1,3 +1,88 @@
+export type BuildError = BasicErrorInfo & {
+  title: string
+  pluginInfo?: string
+  locationInfo?: string
+}
+
+export type BasicErrorInfo = {
+  message: string
+  stack: string
+  severity: string
+  type: ErrorTypes
+  errorInfo: ErrorInfo
+  errorProps: any
+  errorMetadata: any
+  tsConfigInfo?: any
+} & ErrorType
+
+enum ErrorSeverity {
+  success = 'success',
+  none = 'none',
+  info = 'info',
+  warning = 'warning',
+  error = 'Error',
+}
+
+enum StackType {
+  none = 'none',
+  stack = 'stack',
+  message = 'message',
+}
+
+type GroupFunction = ({ location }: { location: ErrorLocation }) => string
+export type TitleFunction = ({ location }: { location: ErrorLocation }) => string
+
+export interface ErrorType {
+  title: TitleFunction | string
+  locationType?: string
+  showErrorProps?: boolean
+  rawStack?: boolean
+  showInBuildLog?: boolean
+  group?: GroupFunction
+  severity: keyof typeof ErrorSeverity
+  stackType: keyof typeof StackType
+}
+
+export type ErrorInfo = {
+  plugin?: any
+  tsConfig?: any
+  location: ErrorLocation
+}
+
+export type BuildCommandLocation = {
+  buildCommand: string
+  buildCommandOrigin: string
+}
+
+export type FunctionsBundlingLocation = {
+  functionName: string
+  functionType: string
+}
+
+export type CoreStepLocation = {
+  coreStepName: string
+}
+
+export type PluginLocation = {
+  event: string
+  packageName: string
+  loadedFrom: string
+  origin: string
+  input?: string
+}
+
+export type APILocation = {
+  endpoint: string
+  parameters?: any
+}
+
+export type ErrorLocation =
+  | BuildCommandLocation
+  | FunctionsBundlingLocation
+  | CoreStepLocation
+  | PluginLocation
+  | APILocation
```

Comment:
This came from analysing:
- https://github.com/netlify/build/blob/481a42e210c9f33176f70dfbfc95b84cbe65f1b0/packages/build/src/error/parse/location.js#L3-L61